src/main/java/uk/gov/pay/adminusers/resources/ServiceResource.java

```diff
@@ -227,44 +227,32 @@ public Response removeUserFromService(@PathParam("serviceExternalId") String ser
 
     @Path("/{serviceExternalId}/stripe-agreement")
     @POST
-    @Produces(APPLICATION_JSON)
     @Consumes(APPLICATION_JSON)
-    public Response createStripeAgreement(@PathParam("serviceExternalId") String serviceExternalId, JsonNode payload) {
-        LOGGER.info("Create stripe agreement POST request - [ {} ]", payload);
-
-        Optional<Errors> maybeErrors = stripeAgreementRequestValidator.validateCreateRequest(payload);
-        if (maybeErrors.isPresent()) {
-            return Response.status(BAD_REQUEST).entity(maybeErrors.get()).build();
-        }
-
-        Optional<ServiceEntity> maybeService = serviceDao.findByExternalId(serviceExternalId);
-        if (!maybeService.isPresent()) {
-            return Response.status(NOT_FOUND).build();
-        }
+    public Response createStripeAgreement(@PathParam("serviceExternalId") String serviceExternalId,
+                                          @NotNull @Valid StripeAgreementRequest stripeAgreementRequest) {
+        LOGGER.info("Create stripe agreement POST request - [ {} ]", stripeAgreementRequest.toString());
 
-        if (stripeAgreementService.findStripeAgreementByServiceId(maybeService.get().getId()).isPresent()) {
-            return Response.status(CONFLICT)
-                    .entity(Errors.from("Stripe agreement information is already stored for this service"))
-                    .build();
+        ServiceEntity service = serviceDao.findByExternalId(serviceExternalId)
+                .orElseThrow( () -> new WebApplicationException(Status.NOT_FOUND));
+        
+        if (stripeAgreementService.findStripeAgreementByServiceId(service.getId()).isPresent()) {
+            throw new WebApplicationException("Stripe agreement information is already stored for this service",
+                    Status.CONFLICT);
         }
 
-        stripeAgreementService.doCreate(
-                maybeService.get().getId(),
-                payload.get(FIELD_IP_ADDRESS).asText(),
-                LocalDateTime.now(ZoneOffset.UTC));
+        stripeAgreementService.doCreate(service.getId(), stripeAgreementRequest, LocalDateTime.now(ZoneOffset.UTC));
```

Comment:
Completely unrelated to the purpose of the PR but I’m surprised we’re using a `LocalDateTime`. We want an absolute time (for which we usually use `ZonedDateTime` though I sometimes think we should probably use `Instant` instead).